core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1142,6 +1142,19 @@ class KafkaApis(val requestChannel: RequestChannel,
     val metadataRequest = request.body[MetadataRequest]
     val requestVersion = request.header.apiVersion
 
+    // Topic IDs are not supported for versions 10 and 11. Topic names can not be null in these versions.
+    if (!metadataRequest.isAllTopics) {
+      metadataRequest.data.topics.forEach{ topic =>
+        // If null, set to the empty string, since the response does not allow null.
+        if (topic.name == null) {
+          topic.setName("")
```

Comment:
It's odd to do this here. Seems like `getErrorResponse` should handle this case in general?

Reply:
We can do that. It should only apply to this specific case, but it will be cleaner there.